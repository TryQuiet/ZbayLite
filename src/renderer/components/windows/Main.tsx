import React, { useEffect } from 'react'
import { Route } from 'react-router-dom'
import classnames from 'classnames'

import Grid from '@material-ui/core/Grid'
import { makeStyles } from '@material-ui/core/styles'

import WindowWrapper from '../ui/WindowWrapper/WindowWrapper'
import Sidebar from '../widgets/sidebar/Sidebar'
import Channel from '../../containers/pages/Channel'

const useStyles = makeStyles(() => ({
  gridRoot: {
    'min-height': '100vh',
    'min-width': '100vw',
    overflow: 'hidden',
    position: 'relative'
  },
  logsContainer: {
    'z-index': 2000,
    position: 'absolute',
    top: 0,
    right: 0
  }
}))

interface MainProps {
  match: {
    url: string
  }
  isLogWindowOpened: boolean
}

export const Main: React.FC<MainProps> = ({
  match,
  isLogWindowOpened
}) => {
  const classes = useStyles({})
  const debounce = (fn, ms: number) => {
    let timer: ReturnType<typeof setTimeout> | null
    return _ => {
      if (timer) {
        clearTimeout(timer)
      }
      timer = setTimeout(_ => {
        timer = null
        fn.apply(this) // eslint-disable-line
      }, ms)
    }
  }
  const [dimensions, setDimensions] = React.useState({
    height: window.innerHeight,
    width: window.innerWidth
  })
  useEffect(() => {
    const debouncedHandleResize = debounce(function handleResize() {
      setDimensions({
        height: window.innerHeight,
        width: window.innerWidth
      })
    }, 1000)

    window.addEventListener('resize', debouncedHandleResize)

    return () => {
      window.removeEventListener('resize', debouncedHandleResize)
    }
  })
  return (
    <>
      <WindowWrapper>
        <Grid container direction='row' className={classes.gridRoot}>
          <Grid item>
            <Sidebar />
          </Grid>
          <Grid item xs>
            <Route path={`${match.url}/channel/:id`} component={Channel} />
            {/* <Route path={`${match.url}/direct-messages/:username`} component={DirectMessages} /> */}
          </Grid>
          {isLogWindowOpened && (
            <Grid
              className={classnames({
                [classes.logsContainer]: dimensions.width <= 900
              })}
              item></Grid>
          )}
        </Grid>
      </WindowWrapper>
    </>
  )
}

export default Main
