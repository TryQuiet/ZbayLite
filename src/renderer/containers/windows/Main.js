import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import * as R from 'ramda'

import MainComponent from '../../components/windows/Main'
import coordinator from '../../store/handlers/coordinator'
import nodeHandlers from '../../store/handlers/node'
import modalsHandlers from '../../store/handlers/modals'
import identitySelectors from '../../store/selectors/identity'
import { fetchBalance } from '../../store/handlers/identity'
import electronStore from '../../../shared/electronStore'

export const mapStateToProps = state => ({
  zecBalance: identitySelectors.balance('zec')(state)
})
export const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      fetch: coordinator.epics.coordinator,
      fetchBalance: fetchBalance,
      disablePowerSleepMode: nodeHandlers.epics.disablePowerSaveMode,
      openSettingsModal: modalsHandlers.actionCreators.openModal(
        'createUsernameModal'
      )
    },
    dispatch
  )
}

export const Main = ({
  zecBalance,
  openSettingsModal,
  fetchBalance,
  ...props
}) => {
  useEffect(() => {
    fetchBalance()
    const isNewUser = electronStore.get('isNewUser')
    const isMigrating = electronStore.get('isMigrating')
    if (isNewUser === true && !isMigrating && zecBalance.gt(0)) {
      openSettingsModal()
    }
  }, [])
  return <MainComponent {...props} />
}

export default R.compose(
  React.memo,
  connect(mapStateToProps, mapDispatchToProps)
)(Main)
