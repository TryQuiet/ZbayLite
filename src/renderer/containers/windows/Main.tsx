import MainComponent from '../../components/windows/Main'

export default MainComponent
