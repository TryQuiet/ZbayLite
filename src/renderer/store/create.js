import { applyMiddleware, createStore, compose } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const composer = (enhancers) => {
  if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      process.env.NODE_ENV === 'development') {
    return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__(enhancers)
  }
  return compose(enhancers)
}

export default ({
  initialState
} = {}) => createStore(
  reducers,
  initialState,
  composer(applyMiddleware(...[errorsMiddleware, createDebounce(), thunk, promise()]))
)
