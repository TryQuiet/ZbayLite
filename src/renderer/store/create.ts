import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'
import createSagaMiddleware from 'redux-saga'
import rootSaga from '../sagas/index.saga'
import { composeWithDevTools } from 'redux-devtools-extension'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const sagaMiddleware = createSagaMiddleware()

export default (initialState = {}) => {
  const store = createStore(
    reducers,
    initialState,
    composeWithDevTools(
      applyMiddleware(...[errorsMiddleware, createDebounce(), sagaMiddleware, thunk, promise()])
    )
  )
  sagaMiddleware.run(rootSaga)
  return store
}
