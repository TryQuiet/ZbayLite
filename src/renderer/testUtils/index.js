import { DateTime, Settings } from 'luxon'
import secp256k1 from 'secp256k1'
import BigNumber from 'bignumber.js'
import { hash } from '../zbay/messages'
import defaultChannels from '../zcash/channels'
import { messageType } from '../../shared/static'
Settings.defaultZoneName = 'utc'

export const vaultTestMessages = [
  {
    id: '123',
    properties: {
      title: 'test',
      createdAt: '123456',
      message: '{"itemId":"6b31f1a5c68902a767eb542fa17daeb338e32d12704ac124ce55994754a5001e","text":"hello"}',
      sender: 'testsender',
      senderUsername: 'testusername',
      type: '11',
      id: '049a7d2ab30817765ea9875bfca40299669e1631329fc58b2a053ff58f78b007'
    }
  },
  {
    id: '1234',
    properties: {
      title: 'test2',
      createdAt: '123456',
      message: '{"itemId":"6b31f1a5c68902a767eb542fa17daeb338e32d12704ac124ce55994754a5001e","text":"hello"}',
      sender: 'testsender',
      senderUsername: 'testusername',
      type: '11',
      id: '149a7d2ab30817765ea9875bfca40299669e1631329fc58b2a053ff58f78b007'
    }
  }
]
export const identities = [
  {
    address: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  {
    address: 'zs1dhqp9dtr4pksnmaynp2k22qduvywejg3neqq4swd4a6gnz6w0m208kefcdm9n2067yn5clcvgsq',
    username: 'Mars'
  }
]

const privateKey = 'ceea41a1c9e91f839c96fba253b620da70992954b2a28b19322b191d8f5e56db'
const pKey = Buffer.alloc(32)
pKey.write(privateKey, 0, 'hex')

export const now = DateTime.utc(2019, 3, 7, 13, 3, 48)

export const createChannel = id => ({
  id: id,
  name: `Channel ${id}`,
  private: Boolean(id % 2),
  address: `zs1testaddress${id}`,
  unread: 12,
  description: id % 2 === 0 ? '' : `Channel about ${id}`,
  advertFee: 0.15,
  keys: {
    ivk: `incoming-viewing-key-${id}`,
    sk: 'test-secret-key'
  }
})

export const createUsersChannel = (id, network = 'testnet') => {
  return {
    id: id,
    name: `Channel ${id}`,
    private: Boolean(id % 2),
    address: defaultChannels.registeredUsers[network].address,
    unread: id,
    description: id % 2 === 0 ? '' : `Channel about ${id}`,
    keys: {
      ivk: `incoming-viewing-key-${id}`
    }
  }
}

export const channels = {
  createChannel,
  createUsersChannel
}

export const createVaultMessage = (
  id,
  createdAt = now.minus({ hours: id }).toSeconds(),
  replyTo = 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya'
) => ({
  id,
  status: 'broadcasted',
  spent: new BigNumber('123'),
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  sender: {
    replyTo,
    username: 'test'
  }
})

export const createMessage = (id, createdAt = now.minus({ hours: id }).toSeconds()) => ({
  id,
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})
export const createItemMessage = (id, createdAt = now.minus({ hours: id }).toSeconds()) => ({
  id,
  type: messageType.BASIC,
  createdAt,
  message: { itemId: 'test1', test: `This is some message ${id}` },
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createReceivedMessage = ({
  id,
  createdAt = now.toSeconds(),
  sender = identities[0]
}) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.BASIC,
  createdAt,
  message: `This is a message with id ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createSendableMessage = ({
  message,
  createdAt = now.toSeconds(),
  type = messageType.BASIC
}) => ({
  type,
  signature: secp256k1.sign(hash(message), pKey).signature,
  createdAt,
  message
})

export const createSendableUserMessage = ({
  message,
  createdAt = now.toSeconds(),
  type = messageType.USER
}) => ({
  type,
  signature: secp256k1.sign(hash(JSON.stringify(message) + createdAt), pKey).signature,
  createdAt,
  message
})

export const createSendableTransferMessage = ({
  message = 'hello',
  createdAt = now.toSeconds()
}) => ({
  type: messageType.TRANSFER,
  spent: new BigNumber('0.32'),
  sender: {
    replyTo: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  receiver: {
    replyTo: 'zs1z9rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Sun'
  },
  createdAt,
  signature: secp256k1.sign(hash(message), pKey).signature,
  message
})

export const createReceivedTransferMessage = ({ id, createdAt = now.toSeconds() }) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.TRANSFER,
  createdAt,
  message: `This is a message with id ${id}`,
  signature: secp256k1.sign(hash(`This is a message with id ${id}`), pKey).signature
})

export const createReceivedUserMessage = ({ id, createdAt = now.toSeconds() }) => ({
  id,
  spent: new BigNumber('0.32'),
  type: messageType.USER,
  createdAt,
  message: {
    firstName: 'testname',
    lastName: 'testlastname',
    nickname: 'nickname',
    address:
      'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
  },
  signature: secp256k1.sign(
    hash(
      JSON.stringify({
        firstName: 'testname',
        lastName: 'testlastname',
        nickname: 'nickname',
        address:
          'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
      }) + id
    ),
    pKey
  ).signature,
  r: secp256k1.sign(
    hash(
      JSON.stringify({
        firstName: 'testname',
        lastName: 'testlastname',
        nickname: 'nickname',
        address:
          'ztestsapling14dxhlp8ps4qmrslt7pcayv8yuyx78xpkrtfhdhae52rmucgqws2zp0zwf2zu6qxjp96lzapsn4r'
      }) + id
    ),
    pKey
  ).recovery
})

export const messages = {
  createReceivedMessage,
  createSendableMessage,
  createReceivedTransferMessage,
  createSendableTransferMessage,
  createMessage,
  createVaultMessage,
  createReceivedUserMessage,
  createSendableUserMessage
}

export const createTransfer = ({ txid, amount = '0', memo = '', change = false, blockHeight = 100 }) => ({
  txid,
  amount,
  memo,
  memohex: memo,
  change,
  blockHeight: 100
})

export const transfers = {
  createTransfer
}

export const createIdentity = ({
  name = 'Saturn',
  address = 'saturn-private-address',
  transparentAddress = 'saturn-transparent-address',
  signerPrivKey = 'AyrXTov+4FEyle1BrxndDhePlUOB1nPV5YvyexC8m6E=',
  signerPubKey = 'Alw9G29ahXm5et9T7FJF3lbXVXNkapR826yPtfMVJYnK',
  keys = {
    tpk: 'saturn-tpk',
    sk: 'saturn-sk'
  },
  shippingData = {
    firstName: 'Saturn',
    lastName: 'the Planet',
    street: 'Coders Dv',
    country: 'Poland',
    region: 'Malopolska',
    city: 'Krakow',
    postalCode: '1337-455'
  }
} = {}) => ({
  name,
  address,
  transparentAddress,
  signerPrivKey,
  signerPubKey,
  shippingData,
  keys
})

export default {
  identities,
  now,
  transfers,
  channels,
  messages,
  createIdentity,
  createMessage,
  createSendableTransferMessage,
  createItemMessage,
  createReceivedMessage,
  vaultTestMessages
}
