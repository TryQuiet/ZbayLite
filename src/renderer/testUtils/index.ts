import { DateTime, Settings } from 'luxon'
import BigNumber from 'bignumber.js'
import { messageType } from '../../shared/static'

Settings.defaultZoneName = 'utc'

export const now = DateTime.utc(2019, 3, 7, 13, 3, 48)

export const identities = [
  {
    address: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  {
    address: 'zs1dhqp9dtr4pksnmaynp2k22qduvywejg3neqq4swd4a6gnz6w0m208kefcdm9n2067yn5clcvgsq',
    username: 'Mars'
  }
]

export const createChannel = (id: number) => ({
  id: id,
  name: `Channel ${id}`,
  private: Boolean(id % 2),
  address: `zs1testaddress${id}`,
  unread: 12,
  description: id % 2 === 0 ? '' : `Channel about ${id}`,
  advertFee: 0.15,
  keys: {
    ivk: `incoming-viewing-key-${id}`,
    sk: 'test-secret-key'
  }
})

export const channels = {
  createChannel
}

export const createVaultMessage = (
  id,
  createdAt = now.minus({ hours: id }).toSeconds(),
  replyTo = 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya'
) => ({
  id,
  status: 'broadcasted',
  spent: new BigNumber('123'),
  type: messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  sender: {
    replyTo,
    username: 'test'
  }
})

export const createMessage = async () => {
  return {
    id: '1',
    type: messageType.BASIC,
    message: 'example message',
    pubKey: 'ownPubKey',
    channelId: 'address',
    createdAt: 123445789,
    signature: 'secretsignature'
  }
}

export const messages = {
  createMessage,
  createVaultMessage
}

export const createIdentity = ({
  name = 'Saturn',
  address = 'saturn-private-address',
  transparentAddress = 'saturn-transparent-address',
  signerPrivKey = 'AyrXTov+4FEyle1BrxndDhePlUOB1nPV5YvyexC8m6E=',
  signerPubKey = 'Alw9G29ahXm5et9T7FJF3lbXVXNkapR826yPtfMVJYnK',
  keys = {
    tpk: 'saturn-tpk',
    sk: 'saturn-sk'
  },
  shippingData = {
    firstName: 'Saturn',
    lastName: 'the Planet',
    street: 'Coders Dv',
    country: 'Poland',
    region: 'Malopolska',
    city: 'Krakow',
    postalCode: '1337-455'
  }
} = {}) => ({
  name,
  address,
  transparentAddress,
  signerPrivKey,
  signerPubKey,
  shippingData,
  keys
})

export default {
  identities,
  now,
  channels,
  messages,
  createIdentity,
  createMessage
}
