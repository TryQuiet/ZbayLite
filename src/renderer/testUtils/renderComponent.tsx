import React, { FC, ReactElement } from 'react'
import { MuiThemeProvider } from '@material-ui/core'
import { Provider } from 'react-redux'

import { render } from '@testing-library/react'

import theme from '../theme'
import store from '../store'
import { Store } from 'redux'

export const renderComponent = (ui: ReactElement, storeState: Store = store): ReturnType<typeof render> => {
  const Wrapper: FC = ({ children }) => (
    <MuiThemeProvider theme={theme}>
      <Provider store={storeState}>
        {children}
      </Provider>
    </MuiThemeProvider>
  )

  return render(ui, { wrapper: Wrapper })
}
