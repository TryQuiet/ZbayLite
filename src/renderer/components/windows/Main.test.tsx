/* eslint import/first: 0 */
import React from 'react'
import { renderComponent } from '../../testUtils/renderComponent'
import { HashRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from '../../store'
import { Main } from './Main'

describe('Main', () => {
  it('renders component', () => {
    const result = renderComponent(
      <HashRouter>
        <Provider store={store}>
          <Main match={{ url: 'test' }} isLogWindowOpened={false} />
        </Provider>
      </HashRouter>
    )
    expect(result.baseElement).toMatchInlineSnapshot(`
      <body>
        <div>
          <div
            class="makeStyles-wrapper-4"
          >
            <div
              class="MuiGrid-root makeStyles-gridRoot-1 MuiGrid-container"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <div
                  class="MuiGrid-root makeStyles-root-108 MuiGrid-container MuiGrid-direction-xs-column"
                >
                  <div
                    class="MuiGrid-root makeStyles-padding-109 MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <div
                        class="makeStyles-root-113"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiButton-root makeStyles-button-114 MuiButton-text"
                          role="button"
                          tabindex="0"
                        >
                          <span
                            class="MuiButton-label makeStyles-buttonLabel-115"
                          >
                            <h4
                              class="MuiTypography-root makeStyles-nickname-116 MuiTypography-h4"
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
                    >
                      <div
                        style="overflow: visible; height: 0px; width: 0px;"
                      >
                        <div
                          class="rc-scrollbars-container"
                          style="position: relative; overflow: hidden; width: 0px; height: 0px;"
                        >
                          <div
                            class="rc-scrollbars-view"
                            style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; overflow: scroll; margin-right: 0px; margin-bottom: 0px;"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item"
                              >
                                <div
                                  class="MuiGrid-root makeStyles-root-187 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
                                >
                                  <div
                                    class="MuiGrid-root MuiGrid-item"
                                  >
                                    <span>
                                      <p
                                        class="MuiTypography-root makeStyles-title-188 makeStyles-clickable-189 MuiTypography-body2"
                                      >
                                        Channels
                                      </p>
                                    </span>
                                  </div>
                                  <div
                                    class="MuiGrid-root MuiGrid-item"
                                  >
                                    <span>
                                      <button
                                        class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-190 MuiIconButton-edgeEnd"
                                        tabindex="0"
                                        type="button"
                                      >
                                        <span
                                          class="MuiIconButton-label"
                                        >
                                          <svg
                                            fill="none"
                                            height="18"
                                            viewBox="0 0 24 24"
                                            width="18"
                                            xmlns="http://www.w3.org/2000/svg"
                                          >
                                            <path
                                              d="M22.0499 12C22.0499 17.5505 17.5504 22.05 12 22.05C6.44949 22.05 1.94995 17.5505 1.94995 12C1.94995 6.44955 6.44949 1.95001 12 1.95001C17.5504 1.95001 22.0499 6.44955 22.0499 12Z"
                                              stroke="white"
                                              stroke-width="1.5"
                                            />
                                            <path
                                              clip-rule="evenodd"
                                              d="M17.3415 12.5982H12.5983V17.3415H11.4018V12.5982H6.65857V11.4018H11.4018V6.65851H12.5983V11.4018H17.3415V12.5982Z"
                                              fill="white"
                                              fill-rule="evenodd"
                                            />
                                          </svg>
                                        </span>
                                        <span
                                          class="MuiTouchRipple-root"
                                        />
                                      </button>
                                    </span>
                                  </div>
                                </div>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item"
                              >
                                <ul
                                  class="MuiList-root"
                                />
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item"
                              >
                                <button
                                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-221"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiButton-label"
                                  >
                                    <div
                                      class="makeStyles-iconDiv-223"
                                    >
                                      <img
                                        src="test-file-stub"
                                      />
                                    </div>
                                    <p
                                      class="MuiTypography-root MuiTypography-body2"
                                    >
                                      Find Channel
                                    </p>
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                            </div>
                            <div
                              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
                            >
                              <div
                                class="MuiGrid-root MuiGrid-item"
                              >
                                <div
                                  class="MuiGrid-root makeStyles-root-187 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
                                >
                                  <div
                                    class="MuiGrid-root MuiGrid-item"
                                  >
                                    <p
                                      class="MuiTypography-root makeStyles-title-188 MuiTypography-body2"
                                    >
                                      Direct Messages
                                    </p>
                                  </div>
                                  <div
                                    class="MuiGrid-root MuiGrid-item"
                                  >
                                    <span>
                                      <button
                                        class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-190 MuiIconButton-edgeEnd"
                                        tabindex="0"
                                        type="button"
                                      >
                                        <span
                                          class="MuiIconButton-label"
                                        >
                                          <svg
                                            fill="none"
                                            height="18"
                                            viewBox="0 0 24 24"
                                            width="18"
                                            xmlns="http://www.w3.org/2000/svg"
                                          >
                                            <path
                                              d="M22.0499 12C22.0499 17.5505 17.5504 22.05 12 22.05C6.44949 22.05 1.94995 17.5505 1.94995 12C1.94995 6.44955 6.44949 1.95001 12 1.95001C17.5504 1.95001 22.0499 6.44955 22.0499 12Z"
                                              stroke="white"
                                              stroke-width="1.5"
                                            />
                                            <path
                                              clip-rule="evenodd"
                                              d="M17.3415 12.5982H12.5983V17.3415H11.4018V12.5982H6.65857V11.4018H11.4018V6.65851H12.5983V11.4018H17.3415V12.5982Z"
                                              fill="white"
                                              fill-rule="evenodd"
                                            />
                                          </svg>
                                        </span>
                                        <span
                                          class="MuiTouchRipple-root"
                                        />
                                      </button>
                                    </span>
                                  </div>
                                </div>
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item"
                              >
                                <ul
                                  class="MuiList-root"
                                />
                              </div>
                              <div
                                class="MuiGrid-root MuiGrid-item"
                              >
                                <button
                                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-221"
                                  tabindex="0"
                                  type="button"
                                >
                                  <span
                                    class="MuiButton-label"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      class="MuiSvgIcon-root makeStyles-icon-222"
                                      focusable="false"
                                      role="presentation"
                                      viewBox="0 0 24 24"
                                    >
                                      <path
                                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                                      />
                                    </svg>
                                    <p
                                      class="MuiTypography-root MuiTypography-body2"
                                    >
                                      New Message
                                    </p>
                                  </span>
                                  <span
                                    class="MuiTouchRipple-root"
                                  />
                                </button>
                              </div>
                            </div>
                          </div>
                          <div
                            class="rc-scrollbars-track rc-scrollbars-track-h"
                            style="position: absolute; right: 2px; bottom: 2px; z-index: 100; border-radius: 3px; left: 2px; height: 6px; display: none;"
                          >
                            <div
                              class="rc-scrollbars-thumb rc-scrollbars-thumb-h"
                              style="position: relative; display: block; height: 100%; cursor: pointer; border-radius: inherit; background-color: rgba(0, 0, 0, 0.2);"
                            />
                          </div>
                          <div
                            class="rc-scrollbars-track rc-scrollbars-track-v"
                            style="position: absolute; right: 2px; bottom: 2px; z-index: 100; border-radius: 3px; top: 2px; width: 6px; display: none;"
                          >
                            <div
                              class="rc-scrollbars-thumb rc-scrollbars-thumb-v"
                              style="position: relative; display: block; height: 100%; cursor: pointer; border-radius: inherit; background-color: rgba(0, 0, 0, 0.2);"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="resize-triggers"
                      >
                        <div
                          class="expand-trigger"
                        >
                          <div
                            style="width: 1px; height: 1px;"
                          />
                        </div>
                        <div
                          class="contract-trigger"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
              />
            </div>
          </div>
        </div>
      </body>
    `)
  })
})
