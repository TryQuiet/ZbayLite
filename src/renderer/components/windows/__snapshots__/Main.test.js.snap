// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders component 1`] = `
<Fragment>
  <Component>
    <WithStyles(ForwardRef(Grid))
      className="gridRoot-mock"
      container={true}
      direction="row"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Memo(WithStyles(Sidebar)) />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={true}
      >
        <Route
          component={[Function]}
          path="test/channel/:id"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(DirectMessages)",
                "type": [Function],
              },
            }
          }
          path="test/direct-messages/:username"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </Component>
</Fragment>
`;
